### DISCUSSION TITLE
CryptographyDeprecationWarning appears when importing oracledb using Python3.6.8

### DISCUSSION THREAD

**Discussion by urosdigital** [oracle/python-oracledb#152] [Q&A] [answered]
Hi,

We recently had to move some scripts to server with details following below.  CryptographyDeprecationWarning appears as stdout when importing oracledb. Is there a way to hide this warning? 
Some scripts are scheduled in crontab and for them a quick and dirty solution to pipe the output is applicable, there are some scripts which needs to be run occasionally and displaying the message is pain in the neck.

warning:
/path_to_my_venv/lib64/python3.6/site-packages/oracledb/connection.py:45: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography (40.0) will be the last to support Python 3.6.
  from . import base_impl, thick_impl, thin_impl

server:
NAME="Red Hat Enterprise Linux"
VERSION="8.7 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.7"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.7 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.7
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.7"

python:
Python 3.6.8

libs:
oracledb           1.2.2
cryptography       39.0.1

Note that i was not able to install .whl since it could not pool target version of cryptography module from oracledb so i had to run pip install oracledb in order to complete installation.

I understand this is a warning coming from cryptography, but it any help would be much appreciated.

BR,
Uros

**Comment by cjbj** [comment, marked as answer]
Does https://cryptography.io/en/3.4.2/faq.html help?

**Reply by urosdigital** [reply]
Hi @cjbj 

Not really but thank you for your time and feedback. 

BR,
Uros

**Reply by cjbj** [reply]
Newer versions of Python are available via AppStream modules.  Can you update?

**Reply by urosdigital** [reply]
Need to check that, it's production server. Hardly IMO.

**Reply by urosdigital** [reply]
After a lots of reading and digging i have fixed it.  In case someone else is having similar issue:

import warnings
warnings.filterwarnings('ignore', message='.*cryptography', ) 
import oracledb

This way the pattern cryptography is used to target the annoying message: 
CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography (40.0) will be the last to support Python 3.6.
from . import base_impl, thick_impl, thin_impl

BR,
Uros

**Reply by cjbj** [reply]
Thanks for sharing!
